Changes to support `auto`

aphillips opened the pull request:
Draft of changes to support `auto` as a value for directional metadata fields, per \[I18N-ACTION-1176\]


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/string-meta/pull/73.html" title="Last updated on Aug 5, 2022, 3:33 PM UTC (cd52a85)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/string-meta/73/afe40ff...aphillips:cd52a85.html" title="Last updated on Aug 5, 2022, 3:33 PM UTC (cd52a85)">Diff</a>

aphillips committed:
Changes to support `auto`

Draft of changes to support `auto` as a value for directional metadata fields, per \[I18N-ACTION-1176\]

r12a commented on index.html:
```diff
@@ -417,24 +414,16 @@ <h2 id="defining_bidi_keywords">Defining Bidirectional Keywords in Specification
 <p>The value <code>rtl</code> indicates a base direction of right-to-left, in exactly the same manner indicated by <a href="https://www.w3.org/TR/css-writing-modes/#direction">CSS writing modes</a> [[CSS-WRITING-MODES-4]]</p>
 
-<p>The value <code>auto</code> indicates that the user agent uses the first strong character of the content to determine the base direction using the <a href="https://html.spec.whatwg.org/multipage/dom.html#the-dir-attribute">algorithm</a> for <code>auto</code> found in [[HTML]].</p>
+<p>The value <code>auto</code> indicates that the user agent uses the <a href="https://html.spec.whatwg.org/multipage/dom.html#the-dir-attribute">algorithm</a> for <code>auto</code> defined by [[HTML]] to determine the base paragraph direction.</p>
```
If we describe first-strong in terms of the bidi algorithm in the next paragraph, why are we describing it in terms of the HTML definition here?

aphillips replied:
Agreed. I split this paragraph up, joining the stuff about the heuristic to the defining paragraph and keeping the guidance on usage separate.

The guidance on usage made me look at section 2.3 again, which I think is now in need of an overhaul 😿

r12a commented on index.html:
```diff
@@ -417,24 +414,16 @@ <h2 id="defining_bidi_keywords">Defining Bidirectional Keywords in Specification
+<p>The value <code>auto</code> indicates that the user agent uses the <a href="https://html.spec.whatwg.org/multipage/dom.html#the-dir-attribute">algorithm</a> for <code>auto</code> defined by [[HTML]] to determine the base paragraph direction.</p>
 
-<p class="note">The heuristic used by <code>auto</code> just looks at the first character with a strong directionality, in a manner analogous to the Paragraph Level determination in the bidirectional algorithm [[UAX9]]. Authors are urged to only use this value as a last resort when the direction of the text is truly unknown and no better server-side heuristic can be applied.</p>
+<p>The heuristic used by <code>auto</code> looks for the first character with a strong directionality, in a manner analogous to the Paragraph Level determination in the bidirectional algorithm [[UAX9]]. When applied to multiple fields or to a document as a whole, it can mean that the direction should be individually derived for each field (with string-local metadata providing an override for case ...
```
How about this:
The heuristic used by auto looks for the first character with a strong directionality, in a manner analogous to the Paragraph Level determination in the bidirectional algorithm [UAX9]. When applied to multiple fields or to a document as a whole, it means that the direction should be individually derived for each field (with string-specific metadata providing an override for cases that cannot be determined automatically). It can be useful for labelling a group of mixed direction strings, when the base direction of most strings can be reliably determined using the first-strong heuristics. Whenever possible, the actual base direction (ltr or rtl) of individual strings should be stored or exchanged instead of auto. Omitting the direction field is preferable when the value is truly unknown.

aphillips replied:
I like this very much. Done.

aphillips replied:
And then I addressed your first comment... by splitting this paragraph up.

r12a commented on index.html:
```diff
@@ -417,24 +414,16 @@ <h2 id="defining_bidi_keywords">Defining Bidirectional Keywords in Specification
+<p>The value <code>auto</code> indicates that the user agent uses the <a href="https://html.spec.whatwg.org/multipage/dom.html#the-dir-attribute">algorithm</a> for <code>auto</code> defined by [[HTML]] to determine the base paragraph direction.</p>
 
-<p class="note">The heuristic used by <code>auto</code> just looks at the first character with a strong directionality, in a manner analogous to the Paragraph Level determination in the bidirectional algorithm [[UAX9]]. Authors are urged to only use this value as a last resort when the direction of the text is truly unknown and no better server-side heuristic can be applied.</p>
+<p>The heuristic used by <code>auto</code> looks for the first character with a strong directionality, in a manner analogous to the Paragraph Level determination in the bidirectional algorithm [[UAX9]]. When applied to multiple fields or to a document as a whole, it can mean that the direction should be individually derived for each field (with string-local metadata providing an override for case ...
```
Please use the attributes `<code class="kw" translate="no">...</code>` for keywords like auto.

aphillips replied:
I made this change globally.

aphillips committed:
Address @r12a's comments

- Add `kw` class and `translate=no` to all `code` elements wrapping keywords.
- Add wording suggestions.
- Additional wording changes to make the text less redundant.

xfq commented:
Merging. Thanks a lot!
